docs/reference/how-to/size-your-shards.asciidoc

```diff
@@ -175,17 +185,20 @@ index                                 prirep shard store
 
 [discrete]
 [[shard-count-recommendation]]
-==== Aim for 20 shards or fewer per GB of heap memory
-
-The number of shards a data node can hold is proportional to the node's heap
-memory. For example, a node with 30GB of heap memory should have at most 600
-shards. The further below this limit you can keep your nodes, the better. If
-you find your nodes exceeding more than 20 shards per GB, consider adding
-another node.
-
-Some system indices for {enterprise-search-ref}/index.html[Enterprise Search]
-are nearly empty and rarely used. Due to their low overhead, you shouldn't
-count shards for these indices toward a node's shard limit.
+==== Aim for 3000 indices or fewer per GB of heap memory on master nodes
+
+The number of indices a master node can manage is proportional to the node's
+heap memory. The exact amount of heap memory each additional index requires
+depends on various factors such as the size of the mapping and the number of
+shards per index. Where possible, the master node will deduplicate the mapping
+metadata across indices with identical mappings. Mapping metadata can be
+deduplicated across indices that are created from the same index template as
+long as they do not use dynamic mapping updates.
+Under the assumption that indices are mostly created identically from a small number of templates
+and no dynamic mapping updates are involved, a good rule of thumb is to aim for 3000 indices or
+fewer per GB of heap on master nodes. For example, if your cluster contains 12,000 indices then
```

Comment:
Can we explicitly call out that the rule of thumb here refers to an index having 1 primary and 1 replica shards?